dirsrvtests/tests/suites/plugins/modrdn_test.py

```diff
@@ -0,0 +1,162 @@
+    """
+    inst = topology_st.standalone
+
+    # Enable automember plugin
+    automember_plugin = AutoMembershipPlugin(inst)
+    automember_plugin.enable()
+
+    # Create definiton for users with A in the name
+    automembers = AutoMembershipDefinitions(inst)
+    automember = automembers.create(
+        properties={
+            "cn": "userswithA",
+            "autoMemberScope": DEFAULT_SUFFIX,
+            "autoMemberFilter": "objectclass=posixAccount",
+            "autoMemberGroupingAttr": "member:dn",
+        }
+    )
+
+    # Create regex rule for users with A in the name
+    automembers_regex_rule = AutoMembershipRegexRules(inst, f"{automember.dn}")
+    automembers_regex_rule.create(
+        properties={
+            "cn": "userswithA",
+            "autoMemberInclusiveRegex": ["cn=.*a.*"],
+            "autoMemberTargetGroup": [f"cn=userswithA,ou=Groups,{DEFAULT_SUFFIX}"],
+        }
+    )
+
+    # Create definiton for users with Z in the name
+    automember = automembers.create(
+        properties={
+            "cn": "userswithZ",
+            "autoMemberScope": DEFAULT_SUFFIX,
+            "autoMemberFilter": "objectclass=posixAccount",
+            "autoMemberGroupingAttr": "member:dn",
+        }
+    )
+
+    # Create regex rule for users with Z in the name
+    automembers_regex_rule = AutoMembershipRegexRules(inst, f"{automember.dn}")
+    automembers_regex_rule.create(
+        properties={
+            "cn": "userswithZ",
+            "autoMemberInclusiveRegex": ["cn=.*z.*"],
+            "autoMemberTargetGroup": [f"cn=userswithZ,ou=Groups,{DEFAULT_SUFFIX}"],
+        }
+    )
+
+    # Enable memberof plugin
+    memberof_plugin = MemberOfPlugin(inst)
+    memberof_plugin.enable()
+
+    # Enable referential integrity plugin
+    referint_plugin = ReferentialIntegrityPlugin(inst)
+    referint_plugin.enable()
+
+    # Restart the instance
+    inst.restart()
+
+    # Create groups
+    groups = Groups(inst, DEFAULT_SUFFIX)
+    groups.create(properties={"cn": "userswithA"})
+    groups.create(properties={"cn": "userswithZ"})
+
+    # Create users
+    users = nsUserAccounts(inst, DEFAULT_SUFFIX)
+
+    # userwitha
+    user_props = USER_PROPERTIES.copy()
+    user_props.update(
+        {
+            "uid": USER_PROPERTIES["uid"] + "a",
+            "cn": USER_PROPERTIES["cn"] + "a",
+        }
+    )
+    user = users.create(properties=user_props)
+
+    # userwithz
+    user_props.update(
+        {
+            "uid": USER_PROPERTIES["uid"] + "z",
+            "cn": USER_PROPERTIES["cn"] + "z",
+        }
+    )
+    user = users.create(properties=user_props)
+
+    # userwithaz
+    user_props.update(
+        {
+            "uid": USER_PROPERTIES["uid"] + "az",
+            "cn": USER_PROPERTIES["cn"] + "az",
+        }
+    )
+    user = users.create(properties=user_props)
+
+    # Rename userwithaz
+    user.rename(new_rdn="uid=userwith")
+    assert user.get_attr_val_utf8("uid") != "userwithaz"
```

Comment:
May be good to also verify that the renamed entry is in the groups userswithA and userswithB. Also checking the for former dn (before rename) is no longer in the groups.

Reply:
In addition, I suggest to run IDM CI tests. freeipa is relying over MO and automember.